odoo/orm/models.py

```diff
@@ -5895,12 +5894,19 @@ def filtered(self, func: str | Callable[[Self], bool]) -> Self:
         if not func:
             # align with mapped()
             return self
-        if isinstance(func, str):
+        if callable(func):
+            # normal function
+            pass
+        elif isinstance(func, str):
             if '.' in func:
-                return self.browse(rec.id for rec in self if any(rec.mapped(func)))
+                return self.browse(rec_id for rec_id, rec in zip(self._ids, self) if any(rec.mapped(func)))
             # avoid costly mapped
             func = self._fields[func].__get__
-        return self.browse(rec.id for rec in self if func(rec))
+        elif isinstance(func, Domain):
+            return func.filter_records(self)
```

Comment:
As `filtered(domain)` bypasses `filtered_domain(domain)`, the override in `analytic.mixin` won't be applied.  IMHO `filtered_domain` should delegate to `filtered_domain(domain)`.